cmd/kubeadm/app/cmd/init.go

```diff
@@ -193,7 +193,15 @@ func newCmdInit(out io.Writer, initOptions *initOptions) *cobra.Command {
 	// sets the data builder function, that will be used by the runner
 	// both when running the entire workflow or single phases
 	initRunner.SetDataInitializer(func(cmd *cobra.Command, args []string) (workflow.RunData, error) {
-		return newInitData(cmd, args, initOptions, out)
+		data, err := newInitData(cmd, args, initOptions, out)
+		if err != nil {
+			return nil, err
+		}
+		// If the flag for skipping phases was empty, use the values from config
+		if len(initRunner.Options.SkipPhases) == 0 {
+			initRunner.Options.SkipPhases = data.cfg.SkipPhases
+		}
```

Comment:
This is one of the few cases in kubeadm where we are assuming that flags and config could be set at the same time for the same field and we are enforcing a precedence rule (flags over config).
This leads me to two considerations/questions:
- Should me make this behavior more clear by documenting it both in the API and on the flags?
- Are we sure this behaviour (flag over config) is consistent across all the flags that can co-exists with config?

Reply:
i'm hoping that long term we can remove flags like --skip-phases. and allow only usage from config.

> Should me make this behavior more clear by documenting it both in the API and on the flags?

i can document it in both places.

> Are we sure this behaviour (flag over config) is consistent across all the flags that can co-exists with config?

yes, one exception is --cri-socket, which for 1.22 we are disallowing as mixed with --config.
another "long term" note here is that criSocket must be a field in KubeletConfiguration and kubeadm should not care about it. have no, crictl vs docker CLI, abstraction and so on.

Reply:
added more comments.